Src/FluentAssertions/Formatting/DateTimeOffsetValueFormatter.cs

```diff
@@ -92,14 +92,7 @@ public void Format(object value, FormattedObjectGraph formattedGraph, Formatting
 
         if (!hasDate && !hasTime)
         {
-            if (HasMilliSeconds(dateTimeOffset))
-            {
-                formattedGraph.AddFragment("0001-01-01 00:00:00." + dateTimeOffset.ToString("fff", CultureInfo.InvariantCulture));
-            }
-            else
-            {
-                formattedGraph.AddFragment("0001-01-01 00:00:00.000");
```

Comment:
You _most definitely_ cannot reach this `if` body, if you have milliseconds set.
Thus, checking inside this `if` whether you have milliseconds set is a bit redundant..